src/main/java/org/bytedeco/javacpp/indexer/Indexer.java

```diff
@@ -116,8 +167,13 @@ protected static final long checkIndex(long i, long size) {
      */
     public long index(long... indices) {
         long index = 0;
+        long[] physicalStrides = physicalStrides(sizes);
         for (int i = 0; i < indices.length && i < strides.length; i++) {
-            index += indices[i] * strides[i];
+            long originalCoordinate = indices[i];
+            long mappedCoordinate = starts[i] 
+                    + strides[i] * (originalCoordinate / blocks[i]) 
+                    + (originalCoordinate % blocks[i]);
+            index += mappedCoordinate * physicalStrides[i];
```

Comment:
This is bound to reduce severely performance when starts, counts, and blocks are not used. You'll need to make sure there are no performance degradation in that case.

Reply:
Yep, right: I tried a different approach now.

Reply:
The JIT compiler can't fully remove the branch, so this will still cause a non-negligible slowdown. The usual way around this in Java is with subclasses.

Reply:
Thanks, very interesting: I guess when the matter just like doing  a `get` on an array/buffer/pointer it is crucial to minimize the overhead.

Reply:
I think the best we could do is create something like an `Index` class with an `index()` method, move the original implementation there, and we could put your new implementation in a `HyperslabIndex` subclass. How does that sound?

Reply:
Actually, it's going to be a bit harder than that. The `index()` method is not called at all for low-dimensional cases to avoid the loop entirely...

Reply:
Ok, I was actually able to get it working pretty well. Please rebase on master and override the methods of `Indexer.Index`.

Reply:
Thanks @saudet! I see now the next natural step: to move `Index` out of `Indexer`!

Thus there will be a default `Index` created just from `sizes` with the default `strides` (row-major contiguous indexing).
Then another `Index` that can allow to customize `strides` (as I see some Presets libraries customize the strides).
And eventually a `HyperslabIndex` which is just another custom `Index`.

Now I see maybe it is misleading in case of `HyperslabIndex` the `sizes` really are part of the `Index`, right?

Reply:
Yes, it probably makes sense to move those fields inside `Index`, and then it wouldn't need to be an inner class...

Reply:
Check the latest push I did: missing step will be to move `sizes` inside `Index`.

Reply:
Let's not break backward compatibility of the public API though.

Reply:
Oh, did I? Let me re-check, that was not supposed to happen. :-)

Reply:
Right, sorry! The methods to get `strides` and protected field as well: but what values should have `strides` in case of an `Index` which is not either `DefaultIndex` or `CustomStridesIndex` though?

Reply:
Throw UnsupportedOperationException?

Reply:
It was temporary... now I think I got it right the whole thing.